### PR TITLE
[SDK] Display plugins on router release page

### PR DESCRIPTION
**Pull request by kumaryash90**

### CODE REVIEW

File: packages/sdk/src/evm/core/classes/contract-publisher.ts [hunk 1]
```diff
@@ -357,11 +358,33 @@ export class ContractPublisher extends RPCConnectionHandler {
       invariant(signer, "A signer is required");
       const publisher = await signer.getAddress();
 
-      const predeployMetadata = await fetchRawPredeployMetadata(
+      const predeployMetadata = await fetchPreDeployMetadata(
```
Comment (joaquim-verges) [inline packages/sdk/src/evm/core/classes/contract-publisher.ts:361]: be careful with this, i remember having a good reason for keeping the `raw` metadata around here, lets keep it as it was

File: packages/sdk/test/evm/before-setup.ts [hunk 2]
```diff
@@ -268,6 +269,13 @@ export const mochaHooks = {
         gasSettings: {
           maxPriceInGwei: 10000,
         },
+        supportedChains: [
+          {
+            chainId: 31337,
+            rpc: ["http://localhost:8545"],
+            nativeCurrency: { name: "ETH", symbol: "ETH", decimals: 18 },
+          },
+        ],
```
Comment (joaquim-verges) [inline packages/sdk/test/evm/before-setup.ts:278]: why do you need this?

File: packages/sdk/test/evm/publisher.test.ts [hunk 3]
```diff
@@ -524,4 +530,110 @@ describe("Publishing", async () => {
...
+    );
+    const compositeAbi = fullMetadata.compositeAbi;
+    expect(
+      compositeAbi != undefined &&
+        compositeAbi.length > MarketplaceV3__factory.abi.length,
+    );
+    // console.log(fullMetadata.compositeAbi);
+  });
+
+  // it("Composite Abi for Plugin Router", async () => {
```
Comment (joaquim-verges) [inline packages/sdk/test/evm/publisher.test.ts:569]: cleanup

**Comment by joaquim-verges** [comment]
@kumaryash90 can you take this to the finish line when you can?

File: packages/sdk/test/evm/before-setup.ts (hunk 2 above)
Comment (kumaryash90) [reply packages/sdk/test/evm/before-setup.ts:278]: we added this for `getChainProvider` to work for the tests

File: packages/sdk/src/evm/core/classes/contract-publisher.ts [hunk 4]
```diff
@@ -370,10 +372,14 @@ export class ContractPublisher extends RPCConnectionHandler {
 
         for (const [network, implementation] of implementationsAddresses) {
           if (implementation !== "") {
+            const compilerMetadata = await fetchContractMetadata(
+              predeployMetadata.metadataUri,
+              this.storage,
```
Comment (joaquim-verges) [inline packages/sdk/src/evm/core/classes/contract-publisher.ts:377]: dont do this inside the for loop, its gonna redownload the same thing over and over. just do it under the raw fetch

Comment (kumaryash90) [reply packages/sdk/src/evm/core/classes/contract-publisher.ts:377]: the loop will break as soon as it finds an implementation address.. so this fetch call will only run once

Comment (joaquim-verges) [reply packages/sdk/src/evm/core/classes/contract-publisher.ts:377]: talked offline, the loop breaks at first hit so its ok, but should prob still be inside the try/catch

**Pull request merged**